src/libopensc/asn1.c

```diff
@@ -577,6 +577,9 @@ static int decode_bit_string(const u8 * inbuf, size_t inlen, void *outbuf,
 
 	if (inlen < 1)
 		return SC_ERROR_INVALID_ASN1_OBJECT;
+	/* ITU-T Rec. X.690 8.6.2.2: The number shall be in the range zero to seven. */
+	if ((*in & ~0x07) != 0)
+		return SC_ERROR_INVALID_ASN1_OBJECT;
```

Comment:
doesn't it also specify that if the bitstring is empty, the unused must be zero?

I think we will need also test coveragate to verify it

Reply:
Yes, indeed. I will add that:

> 8.6.2.3 If the bitstring is empty, there shall be no subsequent octets, and the initial octet shall be zero. 

https://www.itu.int/ITU-T/studygroups/com17/languages/X.690-0207.pdf